addons/project_purchase/models/project_project.py

```diff
@@ -12,36 +12,36 @@ class Project(models.Model):
 
     purchase_orders_count = fields.Integer('# Purchase Orders', compute='_compute_purchase_orders_count', groups='purchase.group_purchase_user', export_string_translation=False)
 
-    @api.depends('analytic_account_id')
     def _compute_purchase_orders_count(self):
-        data = self.env['purchase.order.line']._read_group(
-            [('analytic_distribution', 'in', self.analytic_account_id.ids)],
-            ['analytic_distribution'],
-            ['__count'],
-        )
-        data = {int(account_id): order_count for account_id, order_count in data}
+        purchase_count_per_project = {
+            project.id: count
+            for project, count in self.env['purchase.order']._read_group(
+                [('project_id', 'in', self.ids)],
+                ['project_id'], ['__count'],
+            )
+        }
         for project in self:
-            project.purchase_orders_count = data.get(project.analytic_account_id.id, 0)
+            project.purchase_orders_count = purchase_count_per_project.get(project.id)
 
     # ----------------------------
     #  Actions
     # ----------------------------
 
     def action_open_project_purchase_orders(self):
-        purchase_orders = self.env['purchase.order.line'].search(
-            [('analytic_distribution', 'in', self.analytic_account_id.ids)]
-        ).order_id
+        purchase_orders_domain = [('project_id', '=', self.id)]
         action_window = {
             'name': _('Purchase Orders'),
             'type': 'ir.actions.act_window',
             'res_model': 'purchase.order',
             'views': [[False, 'tree'], [False, 'form']],
-            'domain': [('id', 'in', purchase_orders.ids)],
+            'domain': purchase_orders_domain,
```

Comment:
We could directly write the domain here, no need to create a variable for that. 🙂

Reply:
We use it once more L43 :shrug:

Reply:
Ah shit, I did not see it 🙈
You can indeed ignore my comment then